### DISCUSSION TITLE
transcode DVD .iso

### DISCUSSION THREAD

**Discussion by jtmoon79** [gerbera/gerbera#2677] [Q&A]
Is it possible to present a DVD `.iso` file? What would the `config.xml` entries under `<transcoding>`  look like?

**Comment by whyman** [comment]
The transcoding system matches on mimetype and runs a process, so if you can put something together that does the transcode as you like on the CLI we can show you were to set that command in Gerbera

**Comment by sylgh** [comment]
This looks like it might be an option ; https://vanalboom.org/node/10

**Comment by jtmoon79** [comment]
I followed a similar blog post https://stefanobolli.blogspot.com/2013/02/streaming-dvd-iso-with-mediatomb-and-vlc.html
**However, gerbera is not processing `.iso` files. It appears to ignore them during the directory scan.**

Here is a diff of the additions to `/etc/gerbera/config.xml`
```text
49a50,54
>     <!-- added by JTM -->
>     <custom-http-headers>
>       <add header="transferMode.dlna.org: Streaming"/>
>       <add header="contentFeatures.dlna.org: DLNA.ORG_OP=01;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=01500000000000000000000000000000"/>
>     </custom-http-headers>
79c84,85
---
>         <!-- added by JTM -->
>         <map from="iso" to="video/dvdiso"/>
80a87
109a117
>         <treat mimetype="video/dvdiso" as="dvdiso"/>
143a152,153
>       <!-- added by JTM -->
>       <transcode mimetype="video/dvdiso" using="iso2stream"/>
146c156,164
---
>       <!-- added by JTM -->
>       <profile name="iso2stream" enabled="yes" type="external">
>         <mimetype>video/mpeg</mimetype>
>         <accept-url>yes</accept-url>
>         <first-resource>yes</first-resource>
>         <agent command="/usr/bin/vlc" arguments="-I dummy dvdsimple://%in :sout=#duplicate{dst=std{access=file,mux=ps,dst=%out}}} --sout-all vlc:quit"/>
>         <buffer size="1048576" chunk-size="131072" fill-size="262144"/>
>       </profile>
```

I also added `--debug` to the gerbera service command-line.
Reviewing the `gerbera.log` file:
- shows message `INFO: Configuration check succeeded.`
- shows the directory `/movies` containing symlinked `.iso` files is scanned

When I review the loaded Video files from the web interface (click _Database_ -> _Video_ -> _All Video_) I do not see any `.iso` files listed. I do see other video files like symlinked `.avi` files that reside in the same path `/movies`.

I verified the `.iso` files can be read by user `gerbera`
```shell
$ sudo -u gerbera -- file -L /movies/DVD\ -\ Predator\ \(Special\ Edition\)\ -\ Disc\ 1\ of\ 2.iso
/movies/DVD - Predator (Special Edition) - Disc 1 of 2.iso: UDF filesystem data (version 1.5) 'PREDATOR_SE_DISC1_169'
```

I'm not sure how to force gerbera to load `.iso` files.

**Reply by KarlStraussberger** [reply]
As iso files do not present any media data to ffmpeg or taglib they are not recognized as video files. You have to navigate to PC Directory (which contains everything).

Also you can play around with external nfo file support in current head (likely gerbera 1.11.1)
